### PR TITLE
[release/6.0] Downgrade NuGet packages to 6.0.5

### PR DESCRIPTION
**Pull request by dougbu**
- see https://github.com/dotnet/installer/pull/17517
- need to maintain alignment w/ versions bundled in .NET SDK

### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by MichaelSimons** [comment]
I was expecting some code changes because of the breaking changes between these two versions.  source-build surfaced surfaced these in https://github.com/dotnet/installer/pull/17517.

**Comment by dougbu** [comment]
ugh, I forgot about reversing the changes 😦 that'll have to wait 'til I get back from an appointment this afternoon

**Review by mmitche — DISMISSED** [review]

**Review by mmitche — APPROVED** [review]

File: src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs [hunk 1]
```diff
@@ -37,7 +37,6 @@ private bool IsSigned(string path)
             IEnumerable<ISignatureVerificationProvider> providers = new ISignatureVerificationProvider[] {
                 new IntegrityVerificationProvider(),
                 new SignatureTrustAndValidityVerificationProvider(allowUntrustedRootList: null),
-                new AllowListVerificationProvider(allowList: null),
             };
```
Comment (dougbu) [inline src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:40]: @mmitche, @MichaelSimons any objection to leaving this code mostly as I changed it in #14000❔ I could revert that reaction but the new code works just as well&hellip;

Comment (MichaelSimons) [reply src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:40]: no objections from me.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**